su2/gin-file-server.go

```diff
@@ -0,0 +1,101 @@
+package su2
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"time"
+)
+
+const uploadsPath = "su2/uploads"
+
+func GinFileServer() {
+	flag.StringVar(&port, "p", "8080", "port to listen on")
+	flag.StringVar(&address, "a", "localhost", "address to listen on")
+
+	flag.Parse()
+
+	router := gin.Default()
+
+	server := &http.Server{
+		Addr:              net.JoinHostPort(address, port),
+		Handler:           router,
+		ReadTimeout:       time.Second * 30,
+		ReadHeaderTimeout: time.Second * 2,
+		WriteTimeout:      0,
```

Comment:
`WriteTimeout, IdleTimeout, MaxHeaderBytes` if you don't know what values you want to set don't specify them. The default values is still `0`.